src/node/mini_miner.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NODE_MINI_MINER_H
+#define BITCOIN_NODE_MINI_MINER_H
+
+#include <txmempool.h>
+
+#include <memory>
+#include <optional>
+#include <stdint.h>
+
+namespace node {
+
+// Container for tracking updates to ancestor feerate as we include ancestors in the "block"
+struct MockMempoolEntry {
+    CAmount fee_individual;
+    CAmount fee_with_ancestors;
+    int64_t vsize_individual;
+    int64_t vsize_with_ancestors;
+    const CTransaction& tx;
+
+public:
```

Comment:
All struct members are public by default, so can either drop `public` here (and removed the getters), or make the struct a class if any of the data members above are intended to be private (for instance, `tx` and `fee_individual` have public getters and could be private, or just be public and drop the getter).

At first look it seems some of the data members need to be public, in which case their getters should be used or removed. This builds:

```diff
-struct MockMempoolEntry {
+class MockMempoolEntry
+{
     CAmount fee_individual;
-    CAmount fee_with_ancestors;
-    int64_t vsize_individual;
-    int64_t vsize_with_ancestors;
     const CTransaction& tx;
 
 public:
+    CAmount fee_with_ancestors;
+    int64_t vsize_individual;
+    int64_t vsize_with_ancestors;
     explicit MockMempoolEntry(CTxMemPool::txiter entry) :
         fee_individual{entry->GetModifiedFee()},
         fee_with_ancestors{entry->GetModFeesWithAncestors()},
```

Reply:
I'd say just remove the getters, it's fine to keep these public. This struct is only used by `MiniMiner`.
For background, I originally was trying to align the interface with `CTxMemPoolEntry` like `CTxMemPoolModifiedEntry` to reuse the `CompareTxMemPoolEntryByAncestorFee` comparator in a multi index container, but then realized using simple `std::map`s was enough.